data-access/mandataris-download.ts

```diff
@@ -0,0 +1,257 @@
+    `;
+  }
+
+  if (persoonIds.length >= 1) {
+    const idValues = persoonIds.map((id) => sparqlEscapeString(id)).join('\n');
+    persoonFilter = `
+      VALUES ?persoonId { ${idValues} }
+
+      ?mandataris mandaat:isBestuurlijkeAliasVan ?persoon.
+      ?persoon mu:uuid ?persoonId.
+    `;
+  }
+
+  if (bestuursFunctieCodeIds.length >= 1) {
+    const idValues = bestuursFunctieCodeIds
+      .map((id) => sparqlEscapeString(id))
+      .join('\n');
+    mandaatTypeFilter = `
+      VALUES ?functieCode { ${idValues} }
+      ?mandaat org:role ?bestuursfunctieCode .
+      ?bestuursfunctieCode mu:uuid ?functieCode.
+    `;
+  }
+
+  const queryString = `
+    PREFIX mandaat: <http://data.vlaanderen.be/ns/mandaat#>
+    PREFIX mu: <http://mu.semte.ch/vocabularies/core/>
+    PREFIX lmb: <http://lblod.data.gift/vocabularies/lmb/>
+    PREFIX org: <http://www.w3.org/ns/org#>
+    PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>
+    PREFIX ext: <http://mu.semte.ch/vocabularies/ext/>
+    PREFIX fractieType: <http://data.vlaanderen.be/id/concept/Fractietype/>
+
+
+    SELECT DISTINCT ?mandataris
+    WHERE {
+      ?mandataris a mandaat:Mandataris.
+      ?mandataris org:holds ?mandaat.
+
+      ?bestuursorgaan lmb:heeftBestuursperiode ?bestuursperiode.
+      ?bestuursperiode mu:uuid ${sparqlEscapeString(bestuursperiodeId)}.
+
+      ${bestuursorgaanInTijdFilter ?? ''}
+      ${onlyActiveFilter ?? ''}
+      ${persoonFilter ?? ''}
+      ${getFractieFilters(filters)}
+      ${mandaatTypeFilter ?? ''}
+    }
+  `;
+
+  const sparqlResult = await query(queryString);
+
+  return getSparqlResults(sparqlResult).map((res) => res.mandataris.value);
+}
+
+function getFractieFilters(filters): string {
+  const {
+    fractieIds,
+    hasFilterOnOnafhankelijkeFractie,
+    hasFilterOnNietBeschikbareFractie,
+  } = filters;
+  const idValues = fractieIds.map((id) => sparqlEscapeString(id)).join('\n');
+  const unions = [
+    {
+      apply:
+        fractieIds.length === 0 &&
+        !hasFilterOnNietBeschikbareFractie &&
+        !hasFilterOnOnafhankelijkeFractie,
+      filter: `
+      UNION {
+        OPTIONAL {
+          ?mandataris org:hasMembership ?lidmaatschap.
+          ?lidmaatschap org:organisation ?fractie.
+        }
+      }
+      `,
+    },
+    {
+      apply: hasFilterOnNietBeschikbareFractie,
+      filter: `
+        UNION {
+         FILTER NOT EXISTS {
+           ?mandataris org:hasMembership ?lidmaatschap.
+           ?lidmaatschap org:organisation ?fractie.
+         }
+        }
+      `,
+    },
+    {
+      apply: hasFilterOnOnafhankelijkeFractie,
+      filter: `
+        UNION {
+         ?fractie ext:isFractietype fractieType:Onafhankelijk.
+        }
+      `,
+    },
+  ];
+  const unionFilters = unions
+    .filter((union) => union.apply)
+    .map((union) => union.filter)
+    .join('\n');
+
+  return `
+    {
```

Comment:
i think you officially need to wrap your unions in a {} block like `{ { <clause1> } UNION { <clause2> } }`

Reply:
The outer curlies {} are from the where clause no?

Reply:
no because you may want to do 
```
[...]
WHERE {
  ?s a <thing> .
  { { <clause1> } UNION { <clause2> } }
}

Reply:
Encapsulated the unions